intake_netflow/__init__.py

```diff
@@ -0,0 +1,49 @@
+from dask.bytes import open_files
+
+from intake.source import base
+
+from .v9 import RecordStream
+
+
+class Plugin(base.Plugin):
+    def __init__(self):
+        super(Plugin, self).__init__(name='netflow', version='0.1', container='python', partition_access=False)
+
+    def open(self, urlpath, **kwargs):
+        """
+        Parameters:
+            urlpath : str
+                Location of the data files; can include protocol and glob characters.
+            kwargs : dict
+                Additional parameters to pass to ``intake_netflow.v9.RecordStream``.
+        """
+        base_kwargs, source_kwargs = self.separate_base_kwargs(kwargs)
+        return NetflowSource(urlpath=urlpath, netflow_kwargs=source_kwargs, metadata=base_kwargs['metadata'])
+
+
+class NetflowSource(base.DataSource):
+    def __init__(self, urlpath, netflow_kwargs, metadata):
+        self._init_args = dict(netflow_kwargs=netflow_kwargs, metadata=metadata)
+
+        self._urlpath = urlpath
+        self._netflow_kwargs = netflow_kwargs
+        self._streams = [RecordStream(f) for f in open_files(urlpath, mode='rb')]
```

Comment:
I don't see `with openfile as f:` anywhere, so I don't think this will actually work for any non-local file. The output of `open_files` are `dask.bytes.core.OpenFile` objects, not real open files, so I'm surprised it works even for local files - you cannot `peek` or otherwise interact with them! Perhaps the `with` in `get_partition` is doing magic?

Can you conceive of a non-local test that we can do, perhaps post a small publicly-acessible file to S3?

Reply:
In fact, since we always process a whole file in one go, the creation of the readers can be deferred

```
        self._streams = open_files(urlpath, mode='rb')  # lazy file objects

...
    def _get_partition(self, i):
        with self._streams[i] as f:   # actually opens file
            return list(RecordStream(f))
```

Reply:
I'm surprised it worked also. I added tests for a data source and catalog. This uncovered the problem, so a local test should be sufficient.